docs/workflows/accessing-payload-data.md

```diff
@@ -0,0 +1,330 @@
+# Accessing Payload Data
+
+There are a few different methods for accessing and manipulating data from workflow payloads and device data queries. Depending on the use case, you may run across a point while building your app where you're asked to supply a data reference in one of the following formats.
+
+```json
+{
+  "data" : {
+    "aNumberProp": 42,
+    "aStringProp": "Hello world",
+    "aBooleanProp": true,
+    "aFalsyProp" : 0,
+    "aDecimal": 123.456789,
+    "anUndefinedProp": undefined,
+    "aNullProp": null,
+    "aJavascriptDateObject": Wed Sep 28 2016 15:40:25 GMT-0400 (EDT),
+    "anArray": [
+      44,
+      "Goodbye world",
+      false,
+      {
+        "anObjectPropInAnArray": "I'm deep!"
+      }
+    ],
+    "anObject": {
+      "aNestedArray": [
+        "World Star!",
+        20,
+        { }
+      ],
+      "aNestedNumber": 56,
+      "aNestedString": "Wally World"
+    }
+  }
+}
+```
+
+## Payload Paths
+
+Many workflow nodes, such as the [HTTP node](/workflows/data/http/), the [Get](/workflows/data/get-value/) and [Store Value](/workflows/data/store-value/) nodes and the [Math node](/workflows/logic/math/), require users to define a payload path at which either data will be accessed for evaluation, or at which the returned result of an operation will be stored. In these cases we use **payload paths**.
+
+Payload paths are dot-separated references to object properties, such as `foo.bar`. If any property in the chain returns `undefined`, the value of the payload path will also return undefined.
+
+When accessing an array value by index, the index must be wrapped in square brackets and preceded by a dot, such as  `foo.bar.[1]`. Square brackets can also be used to access specific characters in a string value.
```

Comment:
should add a node about how [ ] are used for paths that have special chars in them, like spaces.  so foo.bar.[ a b ] is valid.

Also, the path of a single . references the entire payload